src/librustdoc/html/format.rs

```diff
@@ -947,8 +950,14 @@ impl clean::FnRetTy {
 }
 
 impl clean::BareFunctionDecl {
-    fn print_generic_params(&self) -> impl fmt::Display + '_ {
-        comma_sep(self.generic_params.iter().map(|g| g.print()))
+    fn print_hrtb_bounds_with_space(&self) -> impl fmt::Display + '_ {
+        display_fn(move |f| {
+            if !self.generic_params.is_empty() {
```

Comment:
This field is poorly named, I had to think why the only possible parameters could be HRTB: it's because `fn` is a function pointer and there's no place to *write* the generic parameter, `fn<T>(a: T)` is invalid syntax like we saw before. As a follow-up, not necessarily here, it would be nice to rename this to `hrtb` or something like that.

Reply:
Well, I guess it's named `generic_params` in the HIR :/ https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.BareFnTy.html this is kind of confusing but if it's used elsewhere I'd rather not mess with it.

Reply:
Marking as resolved since if we change it, it won't be in this PR.